SonarQube.Bootstrapper/Interfaces/IBuildAgentUpdater.cs

```diff
@@ -0,0 +1,32 @@
+﻿//-----------------------------------------------------------------------
+// <copyright file="IBuildAgentUpdater.cs" company="SonarSource SA and Microsoft Corporation">
+//   Copyright (c) SonarSource SA and Microsoft Corporation.  All rights reserved.
+//   Licensed under the MIT License. See License.txt in the project root for license information.
+// </copyright>
+//-----------------------------------------------------------------------
+
+using SonarQube.Common;
+using System;
+
+namespace SonarQube.Bootstrapper
+{
+    /// <summary>
+    /// Component that downloads the SonarQube.MSBuild.Runner binaries from the SonarQube server
+    /// </summary>
+    /// <remarks>This interface was introduced to support testing</remarks>
+    public interface IBuildAgentUpdater
+    {
+        /// <summary>
+        /// Attempts to update the SonarQube.MSBuild.Runner binaries on the local machine
+        /// </summary>
+        /// <param name="hostUrl">Address of the SonarQube server</param>
+        /// <param name="targetDir">Directory to which the new binaries to copied</param>
+        /// <returns>True if the updated succeeded, otherwise false</returns>
+        bool TryUpdate(string hostUrl, string targetDir, ILogger logger);
+
+        /// <summary>
+        /// Verifies that the pre/post-processors are compatible with this version of the bootstrapper
+        /// </summary>
+        bool CheckBootstrapperApiVersion(string versionFilePath, Version bootstrapperVersion);
```

Comment:
This interface was re-added to support testing, which is why CheckBootstrapperApiVersion is a separate method, rather than being handled internally as part of TryUpdate.